### PR TITLE
Add documentation for Elasticsearch scaler

### PR DESCRIPTION
**Pull request by orphaner**
Signed-off-by: Nicolas L <nicolas.lassalle@zenika.com>

Adds the documentation for the new born Elasticsearch scaler

### Checklist

- [X] Commits are signed with Developer Certificate of Origin (DCO)

Related to [#2311](https://github.com/kedacore/keda/pull/2311)

### CODE REVIEW

**Comment by orphaner** [comment]
Documentation: [#586](https://github.com/kedacore/keda-docs/pull/586)

File: content/docs/2.5/scalers/elasticsearch.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+title = "Elasticsearch"
+availability = "v2.5+"
+maintainer = "Community"
+description = "Scale applications based on elasticsearch search template query result."
+layout = "scaler"
++++
+
+### Trigger Specification
+
+This specification describes the `elasticsearch` trigger that scales based on result of an [elasticsearch search template](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-template.html) query. This lets you run searches without exposing Elasticsearch’s query syntax to the keda scaler.
```
Comment (tomkerkhove) [inline content/docs/2.5/scalers/elasticsearch.md:11]:
```suggestion
This specification describes the `elasticsearch` trigger that scales based on the result of an [elasticsearch search template](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-template.html) query.

This lets you run searches without exposing Elasticsearch’s query syntax to the KEDA scaler.
```

File: content/docs/2.5/scalers/elasticsearch.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    metadata:
+      addresses: "http://localhost:9200"
+      username: "elastic"
+      passwordFromEnv: "ELASTIC_PASSWORD"
+      index: "my-index"
+      searchTemplateName: "my-search-template-name"
+      params: "param1:value1;param2:value2"
+      valueLocation: "hits.total.value"
+      targetValue: "1"
+
```
Comment (tomkerkhove) [inline content/docs/2.5/scalers/elasticsearch.md:27]: Nit
```suggestion
```

File: content/docs/2.5/scalers/elasticsearch.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+- `addresses` - Comma separated list of hosts and ports of the Elasticsearch cluster client nodes.
+- `unsafeSsl` - Skip certificate validation when connecting over HTTPS. (Values: `true`, `false`, Default: `false`, Optional).
+- `username` - Username to authenticate with to Elasticsearch cluster.
+- `passwordFromEnv` - Environment variable to read the authentication password from to authenticate with the Elasticsearch cluster.
+- `index` - Comma separated list of indexes to run the search template query on.
+- `searchTemplateName` - The search template name to run.
+- `params` - Parameters that will be used by the search template. It supports multiples params separated by a semicolon character ( `;` )
+- `valueLocation` - [GJSON path notation](https://github.com/tidwall/gjson#path-syntax) to refer to the field in the payload containing the metric value.
+- `targetValue` - Target value to scale on. When the metric provided by the API is equal or higher to this value, KEDA will start scaling out. When the metric is 0 or less, KEDA will scale down to 0.
```
Comment (tomkerkhove) [inline content/docs/2.5/scalers/elasticsearch.md:40]: Moved 2 entries to sort it more logically
```suggestion
- `addresses` - Comma separated list of hosts and ports of the Elasticsearch cluster client nodes.
- `username` - Username to authenticate with to Elasticsearch cluster.
- `passwordFromEnv` - Environment variable to read the authentication password from to authenticate with the Elasticsearch cluster.
- `searchTemplateName` - The search template name to run.
- `targetValue` - Target value to scale on. When the metric provided by the API is equal or higher to this value, KEDA will start scaling out. When the metric is 0 or less, KEDA will scale down to 0.
- `params` - Parameters that will be used by the search template. It supports multiples params separated by a semicolon character ( `;` )
- `valueLocation` - [GJSON path notation](https://github.com/tidwall/gjson#path-syntax) to refer to the field in the payload containing the metric value.
- `unsafeSsl` - Skip certificate validation when connecting over HTTPS. (Values: `true`, `false`, Default: `false`, Optional).
```

File: content/docs/2.5/scalers/elasticsearch.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+**Parameter list:**
+
+- `addresses` - Comma separated list of hosts and ports of the Elasticsearch cluster client nodes.
+- `unsafeSsl` - Skip certificate validation when connecting over HTTPS. (Values: `true`, `false`, Default: `false`, Optional).
+- `username` - Username to authenticate with to Elasticsearch cluster.
+- `passwordFromEnv` - Environment variable to read the authentication password from to authenticate with the Elasticsearch cluster.
+- `index` - Comma separated list of indexes to run the search template query on.
+- `searchTemplateName` - The search template name to run.
+- `params` - Parameters that will be used by the search template. It supports multiples params separated by a semicolon character ( `;` )
```
Comment (tomkerkhove) [inline content/docs/2.5/scalers/elasticsearch.md:38]: `params` or `parameters`? I'm personally not a fan of abbreviations but that's just my opinion

File: content/docs/2.5/scalers/elasticsearch.md (hunk 1 above)
Comment (tomkerkhove) [reply content/docs/2.5/scalers/elasticsearch.md:11]: Does the last line actually matter to end-users?

**Review by tomkerkhove — COMMENTED** [review]
LGTM, added a few remarks/nits

File: content/docs/2.5/scalers/elasticsearch.md (hunk 1 above)
Comment (orphaner) [reply content/docs/2.5/scalers/elasticsearch.md:11]: No. This can be removed.

File: content/docs/2.5/scalers/elasticsearch.md (hunk 4 above)
Comment (orphaner) [reply content/docs/2.5/scalers/elasticsearch.md:38]: same here. But I reused the `params` key used on the [elastisearch searh template](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-template.html#create-search-template)

File: content/docs/2.5/scalers/elasticsearch.md (hunk 3 above)
Comment (orphaner) [reply content/docs/2.5/scalers/elasticsearch.md:40]: looks like the `index` parameter has been removed?

Comment (tomkerkhove) [reply content/docs/2.5/scalers/elasticsearch.md]: Woops, that's not intentional. I'd keep it before the search template name as you have suggested.

File: content/docs/2.5/scalers/elasticsearch.md (hunk 4 above)
Comment (tomkerkhove) [reply content/docs/2.5/scalers/elasticsearch.md:37]: Let's go with parameters then since we don't have to align with elasticsearch

Comment (orphaner) [reply content/docs/2.5/scalers/elasticsearch.md:37]: anyway, this can be `parameters`. Please let me know about your choice ;)

Comment (orphaner) [reply content/docs/2.5/scalers/elasticsearch.md]: done

File: content/docs/2.5/scalers/elasticsearch.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+- `addresses` - Comma separated list of hosts and ports of the Elasticsearch cluster client nodes.
+- `username` - Username to authenticate with to Elasticsearch cluster.
+- `passwordFromEnv` - Environment variable to read the authentication password from to authenticate with the Elasticsearch cluster.
+- `index` - Comma separated list of indexes to run the search template query on.
+- `searchTemplateName` - The search template name to run.
+- `targetValue` - Target value to scale on. When the metric provided by the API is equal or higher to this value, KEDA will start scaling out. When the metric is 0 or less, KEDA will scale down to 0.
+- `parameters` - Parameters that will be used by the search template. It supports multiples params separated by a semicolon character ( `;` )
+- `valueLocation` - [GJSON path notation](https://github.com/tidwall/gjson#path-syntax) to refer to the field in the payload containing the metric value.
+- `unsafeSsl` - Skip certificate validation when connecting over HTTPS. (Values: `true`, `false`, Default: `false`, Optional).
```
Comment (JorTurFer) [inline content/docs/2.5/scalers/elasticsearch.md:39]:
```suggestion
- `unsafeSsl` - Skip certificate validation when connecting over HTTPS. (Values: `true`, `false`, Default: `false`, Optional)
```

File: content/docs/2.5/scalers/elasticsearch.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+**Parameter list:**
+
+- `addresses` - Comma separated list of hosts and ports of the Elasticsearch cluster client nodes.
+- `username` - Username to authenticate with to Elasticsearch cluster.
+- `passwordFromEnv` - Environment variable to read the authentication password from to authenticate with the Elasticsearch cluster.
+- `index` - Comma separated list of indexes to run the search template query on.
+- `searchTemplateName` - The search template name to run.
+- `targetValue` - Target value to scale on. When the metric provided by the API is equal or higher to this value, KEDA will start scaling out. When the metric is 0 or less, KEDA will scale down to 0.
+- `parameters` - Parameters that will be used by the search template. It supports multiples params separated by a semicolon character ( `;` )
```
Comment (JorTurFer) [inline content/docs/2.5/scalers/elasticsearch.md:37]:
```suggestion
- `parameters` - Parameters that will be used by the search template. It supports multiples params separated by a semicolon character ( `;` ).
```

**Review by JorTurFer — CHANGES_REQUESTED** [review]
LGTM!
Only 2 small nits related with the dots

**Review by JorTurFer — APPROVED** [review]

File: content/docs/2.5/scalers/elasticsearch.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
++++
+title = "Elasticsearch"
+availability = "v2.5+"
+maintainer = "Community"
+description = "Scale applications based on elasticsearch search template query result."
+layout = "scaler"
```
Comment (zroubalik) [inline content/docs/2.5/scalers/elasticsearch.md:6]: `go_file` is missing here, check other scalers docs please

**Review by zroubalik — CHANGES_REQUESTED** [review]

File: content/docs/2.5/scalers/elasticsearch.md (hunk 7 above)
Comment (orphaner) [reply content/docs/2.5/scalers/elasticsearch.md:6]: done; this should be added to `archetypes/scaler.md` as well.

**Review by zroubalik — APPROVED** [review]

**Pull request merged**